saleor/order/fetch.py

```diff
@@ -63,3 +70,72 @@ def fetch_order_lines(order: "Order") -> list[OrderLineInfo]:
         )
 
     return lines_info
+
+
+@dataclass
+class DraftOrderLineInfo:
+    line: "OrderLine"
+    variant: "ProductVariant"
+    channel_listing: "ProductVariantChannelListing"
+    discounts: list["OrderLineDiscount"]
+    rules_info: list["VariantPromotionRuleInfo"]
+    channel: "Channel"
+    voucher: Optional["Voucher"] = None
+
+    def get_promotion_discounts(self) -> list["OrderLineDiscount"]:
+        return [
+            discount
+            for discount in self.discounts
+            if discount.type in [DiscountType.PROMOTION, DiscountType.ORDER_PROMOTION]
+        ]
+
+    def get_catalogue_discounts(self) -> list["OrderLineDiscount"]:
+        return [
+            discount
+            for discount in self.discounts
+            if discount.type == DiscountType.PROMOTION
+        ]
+
+
+def fetch_draft_order_lines_info(
+    order: "Order", lines: Optional[Iterable["OrderLine"]] = None
+) -> list[DraftOrderLineInfo]:
+    prefetch_related_fields = [
```

Comment:
Is this prefetch needed here? I just did a glimpse into the code, and I did not catch where are those used here.

Reply:
I changed the way we get `variant_channel_listing` to use prefetched data. Cached fields are also used by `fetch_variant_rules_info` function.